src/openlcb/SimpleStack.cxx

```diff
@@ -138,8 +138,9 @@ void SimpleStackBase::default_start_node()
     }
 #if OPENMRN_HAVE_POSIX_FD 
     {
-        auto *space = new FileMemorySpace(
-            SNIP_DYNAMIC_FILENAME, sizeof(SimpleNodeDynamicValues));
+        auto *space =
+            new FileMemorySpace(configUpdateFlow_.open_file(CONFIG_FILENAME),
```

Comment:
I think this is confusing because it gives the appearance that we open the config file more than once.
We should instead add a get_fd() API to ConfigUpdateFlow. We need to check the returned value for being >= 0.

We should probably also prevent more than one call to ConfigUpdateFlow::open_file because it seems to ignore the  filename in that case.

Reply:
is there an equiv of "ensure_open()" on ConfigUpdateFlow? Perhaps we can add that and have it return the fd as well as open if needed.

Reply:
Perhaps it would also make sense to rework this area slightly further to move the CONFIG_FILENAME/SNIP_DYNAMIC_FILENAME into the constructor for ConfigUpdateFlow and have just the call to ensure_open() (or get_fd()) that returns the active FD for the underlying file.

Reply:
We should not do any file operations (including ::open) in constructors. It makes it unnecessarily complicated to set up applications because the order in which static constructors are executed is undeterministic.

I don't know what "ensure_open" is. Since this ensure_open or get_fd would be called in application initialization and not on demand, the sequence of calls is very deterministic. We don't need to prepare for calls of different sequence. Applications always start with opening or resetting their config file. We also need to make sure not to try to open the file before the application had actually run that code.

Reply:
I have been experimenting with this and the code might be cleanest with the suggestion of passing the filename to ConfigUpdateFlow's constructor. The naming ensure_open() makes the functionality clear. ConfigUpdateFlow::open_file() could check the filename matches in the event that it does get called twice.

Reply:
@balazsracz I had understood the suggestion as the filename would be passed to the constructor and stored but the file would not be opened until requested. Is that poor form?

Reply:
> We should not do any file operations (including ::open) in constructors

I agree, I was implying that only the filename would be passed in for future use by `ensure_open`, `get_fd`, `open`, etc..

At the point that global constructors are called it is not guaranteed that all required dependencies will be available (in the case of esp32 this would be VFS, SPIFFS, etc).

Reply:
Actually, doing it this way means also passing the configuration filename into the constructor for SimpleStackBase and therefore the classes that inherit from it. Perhaps that isn't so bad but at first glance it seems a bit messy.

I'll try some options in the morning.

Reply:
`CONFIG_FILENAME` is already a "global" in the openlcb namespace, it will not need to be passed into SimpleStackBase. You don't pass it in today so you won't need to by shifting the reference to a constructor call.

Reply:
Ah, right you are. There's a lesson for me about working on things late at night.

I have updated the pull-request with the config file path now a parameter to ConfigUpdateFlow's constructor. I have not tested this properly yet but it shows how it might look.

Reply:
I'm not supportive of this direction. We are basically copying data from .rodata to RAM in order to use only once and then never free it. This needs to run in extremely memory constrained environments.

I don't see a reason why the configupdateflow should have the filename outside of the local context of the open_file. I don't see why this open would have to be called more than once. It is part of the initialization of the stack, we don't have any ambiguity about where those calls happen.

Reply:
Fair points @balazsracz and I’m happy to change it back if that’s the preference. Easy to keep open_file and add get_fd. 

I’ll take a look again in the morning.

Reply:
sounds good, thanks!